### PR TITLE
Minor fixes around hwloc in CMake and CI

### PR DESCRIPTION
**Pull request by lukaszstolarczuk**
A small fixes in CMake and CI around hwloc options.

- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -414,12 +414,8 @@ if(UMF_BUILD_BENCHMARKS)
...
 endif()
 
-if(UMF_BUILD_EXAMPLES)
-    if(NOT UMF_DISABLE_HWLOC)
-        add_subdirectory(examples)
-    else()
-        message(WARNING "Examples cannot be build - hwloc disabled")
-    endif()
+if(UMF_BUILD_EXAMPLES AND NOT UMF_DISABLE_HWLOC)
+    add_subdirectory(examples)
```
Comment (PatKamin) [inline CMakeLists.txt:418]: I think you could leave a STATUS message indicating that no examples will be built. This way, if someone sets UMF_BUILD_EXAMPLES and disables hwloc at the same time, they will understand why the examples weren't built.

**Review by PatKamin — APPROVED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -100,16 +100,19 @@ else()
     message(FATAL_ERROR "Unknown OS type")
 endif()
 
-if(NOT DEFINED UMF_HWLOC_REPO)
-    set(UMF_HWLOC_REPO "https://github.com/open-mpi/hwloc.git")
-endif()
+if(UMF_DISABLE_HWLOC)
+    message(WARNING "hwloc is disabled, hence OS provider, memtargets, "
```
Comment (lukaszstolarczuk) [inline CMakeLists.txt:104]: hmm... I'm not sure if this should be a warning - this option is used in llvm project now, so they will see this warning all the time, right...?

Comment (PatKamin) [reply CMakeLists.txt:104]: I think that STATUS suffices just to notify the user. We shouldn't warn on using an option that is legitimate.

Comment (lukaszstolarczuk) [reply CMakeLists.txt:104]: yeah, but on the other hand it's disabling half of our functionality 😉

**Review by bratpiorka — APPROVED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (lukaszstolarczuk) [reply CMakeLists.txt]: I've changed to STATUS

**Review by ldorau — APPROVED** [review]
Rebase please.

Maybe move `UMF_DISABLE_HWLOC` before ` UMF_LINK_HWLOC_STATICALLY` in CMake options?

File: CMakeLists.txt [hunk 3]
```diff
@@ -52,6 +52,10 @@ option(
     UMF_LINK_HWLOC_STATICALLY
     "Link UMF with HWLOC library statically (supported for Linux, MacOS and Release build on Windows)"
     OFF)
+option(
```
[the remark this answers is not in the document]
Comment (lukaszstolarczuk) [reply CMakeLists.txt]: Done.

**Review by PatKamin — APPROVED** [review]

**Pull request merged**